src/vm/checker/typecheck/tests/contracts.rs

```diff
@@ -245,3 +189,74 @@ fn test_bad_map_usage() {
         });
     }
 }
+
+
+#[test]
+fn test_expects() {
+    use vm::checker::type_check;
+    let okay = 
+        "(define-map tokens ((id int)) ((balance int)))
+         (define (get-balance)
+            (let ((balance (expects! 
+                              (get balance (fetch-entry tokens (tuple (id 0)))) 
```

Comment:
Wouldn't this be invalid?  I thought `fetch-entry` evaluated to an `Optional<int>` here, meaning that `get` wouldn't work on it since it's not a tuple?

Reply:
Same applies below in `get-balance-3`.

Reply:
`get` passes Options through, to a certain extent: in this case, it gets an Option<Tuple> and returns an Option<Int>

Reply:
Ah okay -- `get A` takes `Optional<(A, B)>` and outputs `Optional<B>` then?

Reply:
Yep

Reply:
Thanks!